# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/windowwatcher/nsWindowWatcher.cpp
# Commit: 50aa16aa36fb
# Full Hash: 50aa16aa36fb3b808d4116377db79bdc6faaa6fe
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2020-04-08 03:36:50
# Regressor Bug: 1616353
# File Overlap Count: 1
# Description:
#   Bug 1616353 - Part 6.3: Replace many nsILoadContext mutators with assertions, r=smaug
#   
#   Many of these setters are redundant, and duplicate work done elsewhere. As it is
#   no longer correct to set nsILoadContext attributes at these times, they're being
#   replaced with assertions.
# ==============================================================================

diff -r c8c7d1ab9fb4 -r 50aa16aa36fb toolkit/components/windowwatcher/nsWindowWatcher.cpp
--- a/toolkit/components/windowwatcher/nsWindowWatcher.cpp	Tue Apr 07 21:39:14 2020 +0000
+++ b/toolkit/components/windowwatcher/nsWindowWatcher.cpp	Tue Apr 07 21:39:16 2020 +0000
@@ -566,12 +566,12 @@
     return NS_ERROR_UNEXPECTED;
   }
 
-  chromeContext->SetPrivateBrowsing(isPrivateBrowsingWindow);
-  chromeContext->SetRemoteSubframes(isFissionWindow);
+  MOZ_ASSERT(chromeContext->UsePrivateBrowsing() == isPrivateBrowsingWindow);
+  MOZ_ASSERT(chromeContext->UseRemoteSubframes() == isFissionWindow);
 
   // Tabs opened from a content process can only open new windows
   // that will also run with out-of-process tabs.
-  chromeContext->SetRemoteTabs(true);
+  MOZ_ASSERT(chromeContext->UseRemoteTabs());
 
   MaybeDisablePersistence(aFeatures, chromeTreeOwner);
 
